src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -5210,49 +5206,6 @@ public void WarningsErrors()
             parsedArgs.Errors.Verify();
         }
 
-        [Fact]
-        public void WarningVersion()
-        {
-            var parsedArgs = DefaultParse(new string[] { "a.cs" }, WorkingDirectory);
-            parsedArgs.Errors.Verify();
-            Assert.Equal(0m, parsedArgs.CompilationOptions.WarningVersion);
-
-            parsedArgs = DefaultParse(new string[] { "/warnversion", "a.cs" }, WorkingDirectory);
-            parsedArgs.Errors.Verify(
-                Diagnostic(ErrorCode.ERR_SwitchNeedsNumber).WithArguments("warnversion").WithLocation(1, 1)
-                );
-            Assert.Equal(0m, parsedArgs.CompilationOptions.WarningVersion);
-
-            parsedArgs = DefaultParse(new string[] { "/warnversion:", "a.cs" }, WorkingDirectory);
-            parsedArgs.Errors.Verify(
-                Diagnostic(ErrorCode.ERR_SwitchNeedsNumber).WithArguments("warnversion").WithLocation(1, 1)
-                );
-            Assert.Equal(0m, parsedArgs.CompilationOptions.WarningVersion);
-
-            parsedArgs = DefaultParse(new string[] { "/warnversion:-1", "a.cs" }, WorkingDirectory);
-            parsedArgs.Errors.Verify(
-                // Warning version must be greater than or equal to '0'.
-                Diagnostic(ErrorCode.ERR_BadWarningVersion).WithArguments("warnversion").WithLocation(1, 1)
-                );
-            Assert.Equal(0m, parsedArgs.CompilationOptions.WarningVersion);
-
-            parsedArgs = DefaultParse(new string[] { "/warnversion:0", "a.cs" }, WorkingDirectory);
-            parsedArgs.Errors.Verify();
-            Assert.Equal(0m, parsedArgs.CompilationOptions.WarningVersion);
-
-            parsedArgs = DefaultParse(new string[] { "/warnversion:5", "a.cs" }, WorkingDirectory);
-            parsedArgs.Errors.Verify();
-            Assert.Equal(5m, parsedArgs.CompilationOptions.WarningVersion);
-
-            parsedArgs = DefaultParse(new string[] { "/warnversion:5.1", "a.cs" }, WorkingDirectory);
```

Comment:
It would be good to verify behavior of many of these scenarios, replacing '/warnversion' with '/warn'.

Reply:
We do have a bunch of tests for border conditions of the /warn flag elsewhere.

---
In reply to: [458392334](https://github.com/dotnet/roslyn/pull/46193#discussion_r458392334) [](ancestors = 458392334)